bpf/lib/nodeport.h

```diff
@@ -2406,10 +2407,22 @@ nodeport_rev_dnat_ingress_ipv4(struct __ctx_buff *ctx, struct trace_ctx *trace,
 	bool allow_neigh_map = true;
 	bool check_revdnat = true;
 	bool has_l4_header;
+	__u32 *vrf_id __maybe_unused = NULL;
 
 	if (!revalidate_data(ctx, &data, &data_end, &ip4))
 		return DROP_INVALID;
 
+#if defined(ENABLE_SRV6) && defined(IS_BPF_LXC)
+	/* Determine if packet belongs to a VRF before we do NAT.
+	 * This is needed because we determine the VRF membership
+	 * based on the source address of the packet. This should
+	 * be fixed in the future by determining the VRF membership
+	 * based on the IP address-agnostic way such as ingress
+	 * interface index.
+	 */
+	vrf_id = srv6_lookup_vrf4(ip4->saddr, ip4->daddr);
+#endif
+
```

Comment:
Can / should this move further below, *after* the `if (!check_revdnat)` condition?

Reply:
Yep, makes sense for me.